modules/apps/transfer/keeper/migrations_test.go

```diff
@@ -0,0 +1,108 @@
+package keeper_test
+
+import (
+	"fmt"
+
+	transferkeeper "github.com/cosmos/ibc-go/v4/modules/apps/transfer/keeper"
+	transfertypes "github.com/cosmos/ibc-go/v4/modules/apps/transfer/types"
+)
+
+func (suite *KeeperTestSuite) TestMigrator_MigrateTraces() {
+
+	testCases := []struct {
+		msg            string
+		malleate       func()
+		expectedTraces transfertypes.Traces
+	}{
+
+		{
+			"success: two slashes in base denom",
+			func() {
+				suite.chainA.GetSimApp().TransferKeeper.SetDenomTrace(
+					suite.chainA.GetContext(),
+					transfertypes.DenomTrace{
+						BaseDenom: "1", Path: "transfer/channel-0/gamm/pool",
+					})
+			},
+			transfertypes.Traces{
+				{
+					BaseDenom: "gamm/pool/1", Path: "transfer/channel-0",
+				},
+			},
+		},
+		{
+			"success: one slash in base denom",
+			func() {
+				suite.chainA.GetSimApp().TransferKeeper.SetDenomTrace(
+					suite.chainA.GetContext(),
+					transfertypes.DenomTrace{
+						BaseDenom: "", Path: "transfer/channel-149/erc/0x85bcBCd7e79Ec36f4fBBDc54F90C643d921151AA",
+					})
+			},
+			transfertypes.Traces{
+				{
+					BaseDenom: "erc/0x85bcBCd7e79Ec36f4fBBDc54F90C643d921151AA/", Path: "transfer/channel-149",
+				},
+			},
+		},
+		{
+			"success: multiple slashes in a row in base denom",
+			func() {
+				suite.chainA.GetSimApp().TransferKeeper.SetDenomTrace(
+					suite.chainA.GetContext(),
+					transfertypes.DenomTrace{
+						BaseDenom: "1", Path: "transfer/channel-5/gamm//pool",
+					})
+			},
+			transfertypes.Traces{
+				{
+					BaseDenom: "gamm//pool/1", Path: "transfer/channel-5",
+				},
+			},
+		},
+		{
+			"success: multihop base denom",
+			func() {
+				suite.chainA.GetSimApp().TransferKeeper.SetDenomTrace(
+					suite.chainA.GetContext(),
+					transfertypes.DenomTrace{
+						BaseDenom: "transfer/channel-1/uatom", Path: "transfer/channel-0",
+					})
+			},
+			transfertypes.Traces{
+				{
+					BaseDenom: "uatom", Path: "transfer/channel-0/transfer/channel-1",
+				},
+			},
+		},
+		{
+			// the expected value will change with the implementation of the fix for #1698
+			"no change: non-standard port",
+			func() {
+				suite.chainA.GetSimApp().TransferKeeper.SetDenomTrace(
+					suite.chainA.GetContext(),
+					transfertypes.DenomTrace{
+						BaseDenom: "customport/channel-7/uatom", Path: "transfer/channel-0/transfer/channel-1",
+					})
+			},
+			transfertypes.Traces{
+				{
+					BaseDenom: "customport/channel-7/uatom", Path: "transfer/channel-0/transfer/channel-1",
+				},
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		suite.Run(fmt.Sprintf("case %s", tc.msg), func() {
+			suite.SetupTest() // reset
+			tc.malleate()     // explicitly set up denom traces
+			migrator := transferkeeper.NewMigrator(suite.chainA.GetSimApp().TransferKeeper)
+			err := migrator.MigrateTraces(suite.chainA.GetContext())
+			suite.Require().NoError(err)
```

Comment:
nit: I think some kind of logical spacing is nice for readability, but not required! :) 

```suggestion
			suite.SetupTest() // reset
			
			tc.malleate() // explicitly set up denom traces
			
			migrator := transferkeeper.NewMigrator(suite.chainA.GetSimApp().TransferKeeper)
			err := migrator.MigrateTraces(suite.chainA.GetContext())
			suite.Require().NoError(err)
```